twisted/conch/ssh/keys.py

```diff
@@ -782,18 +790,24 @@ def public(self):
         return Key(self._keyObject.public_key())
 
 
-    def fingerprint(self):
+    def fingerprint(self, format='md5-hex'):
```

Comment:
[As before](https://github.com/twisted/twisted/pull/468#discussion_r73852166), I feel like this works better as an enumerated type, rather than using ad-hoc string formatting.

Now that you've swapped to declaring both the hash function and the serialization method, however, we can stop using values constants and just use [named constants](https://twistedmatrix.com/documents/current/core/howto/constants.html#constant-names):

```python
import hashlib
from twisted.python.constants import NamedConstant, Names

class FingerprintFormats(Names):
    """
    Constants representing the supported forms of key fingerprints.
    """
    MD5_HEX = NamedConstant()
    SHA256_BASE64 = NamedConstant()
```

Any reason not to do this?